Fails to build from source with GCC 13

### Describe the bug

Current code (0da0f1537ec73f5683f6ca0d08b1e3a3817577da) fails to build from source on GCC 13, e.g. Ubuntu 23.10.

### To reproduce

Run the build as documented on Ubuntu 23.10 machine (or any other that has GCC 13).

### Additional context

Build log:

```
root@acb12774a011:~/nginx-otel/build# make
...
[  4%] Linking CXX static library libabsl_random_seed_sequences.a
[  4%] Built target absl_random_seed_sequences
[  4%] Building CXX object _deps/grpc-build/third_party/abseil-cpp/absl/strings/CMakeFiles/absl_str_format_internal.dir/internal/str_format/arg.cc.o
[  4%] Building CXX object _deps/grpc-build/third_party/abseil-cpp/absl/strings/CMakeFiles/absl_str_format_internal.dir/internal/str_format/bind.cc.o
[  5%] Building CXX object _deps/grpc-build/third_party/abseil-cpp/absl/strings/CMakeFiles/absl_str_format_internal.dir/internal/str_format/extension.cc.o
In file included from /root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.cc:16:
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:34:6: warning: elaborated-type-specifier for a scoped enum must not use the 'class' keyword
   34 | enum class FormatConversionChar : uint8_t;
      | ~~~~ ^~~~~
      |      -----
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:34:33: error: found ':' in nested-name-specifier, expected '::'
   34 | enum class FormatConversionChar : uint8_t;
      |                                 ^
      |                                 ::
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:34:12: error: 'FormatConversionChar' has not been declared
   34 | enum class FormatConversionChar : uint8_t;
      |            ^~~~~~~~~~~~~~~~~~~~
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:35:6: warning: elaborated-type-specifier for a scoped enum must not use the 'class' keyword
   35 | enum class FormatConversionCharSet : uint64_t;
      | ~~~~ ^~~~~
      |      -----
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:35:36: error: found ':' in nested-name-specifier, expected '::'
   35 | enum class FormatConversionCharSet : uint64_t;
      |                                    ^
      |                                    ::
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:35:12: error: 'FormatConversionCharSet' has not been declared
   35 | enum class FormatConversionCharSet : uint64_t;
      |            ^~~~~~~~~~~~~~~~~~~~~~~
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:131:6: warning: elaborated-type-specifier for a scoped enum must not use the 'class' keyword
  131 | enum class Flags : uint8_t {
      | ~~~~ ^~~~~
      |      -----
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:131:18: error: found ':' in nested-name-specifier, expected '::'
  131 | enum class Flags : uint8_t {
      |                  ^
      |                  ::
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:131:12: error: 'Flags' has not been declared
  131 | enum class Flags : uint8_t {
      |            ^~~~~
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:131:28: error: expected unqualified-id before '{' token
  131 | enum class Flags : uint8_t {
      |                            ^
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:143:11: error: 'Flags' does not name a type
  143 | constexpr Flags operator|(Flags a, Flags b) {
      |           ^~~~~
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:147:30: error: 'Flags' was not declared in this scope
  147 | constexpr bool FlagsContains(Flags haystack, Flags needle) {
      |                              ^~~~~
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:147:46: error: 'Flags' was not declared in this scope
  147 | constexpr bool FlagsContains(Flags haystack, Flags needle) {
      |                                              ^~~~~
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:147:58: error: expression list treated as compound expression in initializer [-fpermissive]
  147 | constexpr bool FlagsContains(Flags haystack, Flags needle) {
      |                                                          ^
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:152:27: error: 'Flags' was not declared in this scope
  152 | std::string FlagsToString(Flags v);
      |                           ^~~~~
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:154:51: error: 'Flags' has not been declared
  154 | inline std::ostream& operator<<(std::ostream& os, Flags v) {
      |                                                   ^~~~~
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h: In function 'std::ostream& absl::lts_20211102::str_format_internal::operator<<(std::ostream&, int)':
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:155:29: error: no match for call to '(std::string {aka std::__cxx11::basic_string<char>}) (int&)'
  155 |   return os << FlagsToString(v);
      |                ~~~~~~~~~~~~~^~~
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h: At global scope:
/root/nginx-otel/build/_deps/grpc-src/third_party/abseil-cpp/absl/strings/internal/str_format/extension.h:187:8: warning: elaborated-type-specifier for a scoped enum must not use the 'class' keyword
  187 |   enum class Enum : uint8_t {
      |   ~~~~ ^~~~~
      |        -----
...
```

--- a/.github/workflows/nginx-otel-module-check.yml
+++ b/.github/workflows/nginx-otel-module-check.yml
@@ -24,7 +24,8 @@ jobs:
       - name: Build module
         working-directory: build
         run: |
-          cmake -DNGX_OTEL_NGINX_BUILD_DIR=${PWD}/../nginx/objs ..
+          cmake -DNGX_OTEL_NGINX_BUILD_DIR=${PWD}/../nginx/objs \
+                -DNGX_OTEL_DEV=ON ..
           make -j 4
           strip ngx_otel_module.so
       - name: Archive module
@@ -75,11 +76,11 @@ jobs:
         working-directory: nginx
         run: |
           auto/configure --with-compat --with-debug --with-http_ssl_module \
-          --with-http_v2_module --with-http_v3_module
+                         --with-http_v2_module --with-http_v3_module
           make -j 4
       - name: Run tests
         working-directory: tests
         run: |
           PERL5LIB=../nginx-tests/lib TEST_NGINX_UNSAFE=1 \
-          TEST_NGINX_VERBOSE=1 TEST_NGINX_GLOBALS="load_module \
-          ${PWD}/../build/ngx_otel_module.so;" prove -v .
+              TEST_NGINX_VERBOSE=1 TEST_NGINX_GLOBALS="load_module \
+              ${PWD}/../build/ngx_otel_module.so;" prove -v .

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,7 @@ if(NGX_OTEL_FETCH_DEPS)
     FetchContent_Declare(
         grpc
         GIT_REPOSITORY https://github.com/grpc/grpc
-        GIT_TAG        02384e39185f109bd299eb8482306229967dc970 # v1.46.7
+        GIT_TAG        e241f37befe7ba4688effd84bfbf99b0f681a2f7 # v1.49.4
         GIT_SUBMODULES third_party/protobuf third_party/abseil-cpp third_party/re2
         GIT_SHALLOW    ON)
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,13 +22,12 @@ if(NGX_OTEL_FETCH_DEPS)
         grpc
         GIT_REPOSITORY https://github.com/grpc/grpc
         GIT_TAG        02384e39185f109bd299eb8482306229967dc970 # v1.46.7
-        GIT_SUBMODULES third_party/protobuf third_party/abseil-cpp
+        GIT_SUBMODULES third_party/protobuf third_party/abseil-cpp third_party/re2
         GIT_SHALLOW    ON)
 
     set(gRPC_USE_PROTO_LITE ON  CACHE INTERNAL "")
     set(gRPC_INSTALL        OFF CACHE INTERNAL "")
     set(gRPC_CARES_PROVIDER package CACHE INTERNAL "")
-    set(gRPC_RE2_PROVIDER   package CACHE INTERNAL "")
     set(gRPC_SSL_PROVIDER   package CACHE INTERNAL "")
     set(gRPC_ZLIB_PROVIDER  package CACHE INTERNAL "")
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,8 +6,9 @@ set(NGX_OTEL_NGINX_BUILD_DIR ""
 set(NGX_OTEL_NGINX_DIR "${NGX_OTEL_NGINX_BUILD_DIR}/.."
     CACHE PATH "Nginx source dir")
 
-set(NGX_OTEL_FETCH_DEPS ON CACHE BOOL "Download dependencies")
-set(NGX_OTEL_PROTO_DIR  "" CACHE PATH "OTel proto files root")
+set(NGX_OTEL_FETCH_DEPS ON  CACHE BOOL "Download dependencies")
+set(NGX_OTEL_PROTO_DIR  ""  CACHE PATH "OTel proto files root")
+set(NGX_OTEL_DEV        OFF CACHE BOOL "Enforce compiler warnings")
 
 if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE RelWithDebInfo)
@@ -99,7 +100,9 @@ add_custom_command(
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-add_compile_options(-Wall -Wtype-limits -Werror)
+if (NGX_OTEL_DEV)
+    add_compile_options(-Wall -Wtype-limits -Werror)
+endif()
 
 add_library(ngx_otel_module MODULE
     src/http_module.cpp

